### PR TITLE
Adyen provider setup

### PR DESCRIPTION
**Pull request by jwm0**
QA instructions:

1. Create a new checkout and save it's `token`
2. Open it in the checkout deployment with `?checkoutToken=<token>` appended at the end
3. Complete checkout details (billing, shipping and payment method if required) and click pay
4. Adyen payment page should now be opened up
5. Complete the payment using a test card (see here: https://docs.adyen.com/development-resources/testing/test-card-numbers#visa)
6. The user should be redirected back to checkout with proper order confirmation view (the payment status will be unpaid for now)

### CODE REVIEW

File: apps/checkout-app/backend/payments/providers/adyen/index.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+});
+
+const checkout = new CheckoutAPI(client);
+
+export const createAdyenPayment = async (
+  data: OrderFragment,
+  redirectUrl: string
+) => {
+  const { url } = await checkout.paymentLinks({
+    amount: {
```
Comment (bmigirl) [inline apps/checkout-app/backend/payments/providers/adyen/index.ts:26]: Util function for getting this parsed data

File: apps/checkout-app/backend/payments/providers/adyen/index.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        },
+        reference: pspReference,
+        availableActions: mapAvailableActions(operations),
+      },
+    };
+  }
+
+  if (
+    eventCode ===
+    Types.notification.NotificationRequestItem.EventCodeEnum.Authorisation
```
Comment (bmigirl) [inline apps/checkout-app/backend/payments/providers/adyen/index.ts:111]: Seems to me like this condition is the same as the one above

File: apps/checkout-app/backend/payments/providers/adyen/utils.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import { Types } from "@adyen/api-library";
+
+export const getAdyenAmountFromSaleor = (float: number) => float * 100;
+export const getSaleorAmountFromAdyen = (integer: number) => integer / 100;
+
+export const mapAvailableActions = (
+  operations: Types.notification.NotificationRequestItem.OperationsEnum[]
+): TransactionActionEnum[] =>
+  operations.map((operation) => {
+    if (
```
Comment (bmigirl) [inline apps/checkout-app/backend/payments/providers/adyen/utils.ts:11]: How bout le `switch`

**Review by bmigirl — CHANGES_REQUESTED** [review]

**Review by orzechdev — APPROVED** [review]

File: apps/checkout-app/backend/payments/providers/adyen/utils.ts (hunk 3 above)
Comment (jwm0) [reply apps/checkout-app/backend/payments/providers/adyen/utils.ts:13]: I prefer if as it's less likely to mess up and go into another case when return is omitted, TS here checks that all conditions are met

**Review by szczecha — APPROVED** [review]

**Pull request merged**